sdk/tests/utilities/test_token.py

```diff
@@ -0,0 +1,39 @@
+import asyncio
+import functools
+
+import asynctest
+from functools import wraps, partial
+
+from lusid.utilities import ApiClientFactory
+from utilities import CredentialsSource
+
```

Comment:
On a separate note, I've just noticed that the RefreshingToken does NOT update the refresh token when it refreshes the access token. This means that in time the refresh token will be invalid and it will not longer be able to refresh. This needs to be added in as well.

Reply:
```
async def create_token(self):
    original_token, refresh_token = tu.get_okta_tokens(CredentialsSource.secrets_path())

    refreshed_token = RefreshingToken(token_url=self.config.token_url,
                                      client_id=self.config.client_id,
                                      client_secret=self.config.client_secret,
                                      initial_access_token=original_token,
                                      initial_token_expiry=1,  # 1s expiry
                                      refresh_token=refresh_token,
                                      expiry_offset=3599)  # set to 1s expiry

    return refreshed_token

@run_async
def use_token(self, refresh_token):

    return getattr(refresh_token, "data")

async def test_token_refresh(self):

    refreshing_token = await self.create_token()
    time.sleep(2)
    tokens = await asyncio.gather(*[self.use_token(refreshing_token) for _ in range(0, 5)], return_exceptions=False)
    self.assertEqual(1, len(set(tokens)), "tokens not all equal")
```

Reply:
I added that test and it shows that when different threads try to update the access token simultaneously they all come back with different updated tokens. This means a call to Okta for every thread. Furthermore if the refresh token changes (which it doesn't in this test, I'm not sure when the refresh token gets updated? After 24 hours perhaps?) then I think that some of these requests to refresh the access token will throw an invalid access token error, which I think we have actually seen in some of the data loaders before (which is probably due to the refresh token not being updated as mentioned above, but will also be caused by this issue now all threads use the same access token even if the refresh token is being updated).